### PR TITLE
update developer docs

### LINKED ISSUE #31: Help compiling for development on OSX
I noticed in #11 you had some discussion about compiling on OSX. I was happily using the `Conda-forge` distribution, but now I'm interested in contributing and thus want to compile `capytaine` locally.

I had some trouble compiling locally using the `Makefile` and `setup.py` as directed in the [documentation](https://ancell.in/capytaine/latest/developer_manual/installation.html), until I set the following to make `distutils` call my `homebrew` installed compilers instead the ones from `Xcode` (not sure if all of these are necessary).

```bash
export CC=/usr/local/bin/gcc-9
export CXX=/usr/local/bin/g++
export CPP=/usr/local/bin/cpp
export LD=/usr/local/bin/gcc-9
alias c++=/usr/local/bin/c++
alias g++=/usr/local/bin/g++
alias gcc=/usr/local/bin/gcc-9
alias cpp=/usr/local/bin/cpp
alias ld=/usr/local/bin/gcc-9
alias cc=/usr/local/bin/gcc-9
```

After running compiling successfully, I went to test the installation with `python -m pytest` and realized I needed the additional following packages: `quadpy`, `hypothesis`, `matplotlib`, `vtk` (couldn't get this to work, probably similar to #28).

Anyhow, I was about to ask for help, but figured it out and thought maybe someone else might find this useful.

EDIT: actually the `vtk` functionality works fine. The error in `test_ui_vtk.py` was caused by the line `animation.embed_in_notebook()`- so you must have `ipython` as well.

https://github.com/mancellin/capytaine/blob/4fb70c71bd791272cca548a4ea0c92b24a49622f/pytest/test_ui_vtk.py#L13

### PR DESCRIPTION
**Pull request by ryancoe**
This adds instructions for compiling on macOS (fixes #31) and makes a couple other minor improvements/updates to the [Installation for developers](https://ancell.in/capytaine/latest/developer_manual/installation.html#installation-for-developers) documentation page.

### CODE REVIEW

File: docs/developer_manual/installation.rst [hunk 1]
```diff
@@ -3,7 +3,7 @@ Installation for developers
 ===========================
 
 Capytaine requires **Python 3.6** or higher.
-It has been successfully tested on Python 3.6 and 3.7, and Numpy 1.15 and 1.16.
+It has been successfully tested on Python 3.6, 3.7, and 3.9, and Numpy 1.15, 1.16, and 1.20.
```
Comment (mancellin) [inline docs/developer_manual/installation.rst:6]:
```suggestion
The latest version is tested on Python 3.7 to 3.9, and Numpy 1.17 to 1.19.
```
Cf. current settings of the CI in `.travis.yml`.

File: docs/developer_manual/installation.rst [hunk 2]
```diff
@@ -43,3 +55,12 @@ Add the current directory to the Python path with the following command::
 (This last two commands are included into the ``Makefile`` of the main directory.
 They can be run by ``make develop``.
 The command ``make clean`` deletes the binaries.)
+
+You will also likely want to install Capytaine's optional dependencies::
+
+	conda install matplotlib vtk pytest quadpy
+	pip install pygmsh
```
Comment (mancellin) [inline docs/developer_manual/installation.rst:62]: Also `ipython` and `hypothesis` to run all the tests?

**Review by mancellin — COMMENTED** [review]
Great initiative!

File: docs/developer_manual/installation.rst (hunk 2 above)
Comment (ryancoe) [reply docs/developer_manual/installation.rst]: right! resolved w. 109f63dc5949a95e188f5c9afa78e5c4feb512d2

**Pull request merged**